### PR TITLE
Add automatic creation of GitHub repositories

### PR DESCRIPTION
**Pull request by Keno**
Works fine, though deployment from travis to github is failing: https://travis-ci.org/Keno/IConvBuilder/builds/306624669
I might be doing the encryption wrong.

### CODE REVIEW

File: src/Wizard.jl [hunk 1]
```diff
@@ -167,11 +175,16 @@ end
 Given a filename, normalize it, stripping out extensions.  E.g. the file path
 `"foo/libfoo.tar.gz"` would get mapped to `"libfoo"`.
 """
-function normalize_name(file::AbstractString)
+function normalize_name(file::AbstractString, )
```
Comment (staticfloat) [inline src/Wizard.jl]: I'm not sure what this extra comma is going to do?

File: src/Wizard.jl [hunk 2]
```diff
@@ -1145,25 +1139,353 @@ function step7(state::WizardState)
...
+        t.c_lflag &= ~ECHO
+    end
+    ccall(:ioctl, Cint, (Cint, Cint, Ref{termios}), fd, TCSETS, t)
+    old
+end
+
+const travis_headers = Dict(
+    "Accept" => "application/vnd.travis-ci.2+json",
+    "Content-Type" => "application/json",
+    "User-Agent" => "Travis-WHY-YOU-HAVE-BAD-DOCS (https://github.com/travis-ci/travis-ci/issues/5649)"
```
Comment (staticfloat) [inline src/Wizard.jl]: wow

File: src/Wizard.jl [hunk 3]
```diff
@@ -1145,25 +1139,353 @@ function step7(state::WizardState)
...
+        common_git_repo_setup(repo_dir, repo, state)
+        # Create the README file
+        open(joinpath(repo_dir, "README.md"), "w") do f
+            print(f, """
+            # $github_name
+            
+            [![Build Status](https://travis-ci.org/$(user)/$(github_name).svg?branch=master)](https://travis-ci.org/$(user)/$(github_name))
+            
+            This repository builds binary artifacts for the $(state.name) project. Binary artifacts are automatically uploaded to
+            [this repository's GitHub releases page](https://github.com/staticfloat/ZlibBuilder/releases) whenever a tag is created
```
Comment (staticfloat) [inline src/Wizard.jl]: One more link to `$(user)/$(github_name)`-ify

File: src/Wizard.jl [hunk 4]
```diff
@@ -1145,25 +1139,353 @@ function step7(state::WizardState)
...
+        # Create README file
+        open(joinpath(repo_dir, "README.md"), "w") do f
+            print(f, """
+            # $(state.name) builder
+            
+            This repository builds binary artifacts for the $(state.name) project.
+            This repository has a default .travis.yml file that can be used to build
+            binary artifacts on Travis CI. You will however need to setup the release
+            upload manually. See https://docs.travis-ci.com/user/deployment/releases/.
+            """)
```
Comment (staticfloat) [inline src/Wizard.jl]: Let's also add a small paragraph that says "If you don't want to use Travis, you can build locally then just manually upload to a hosting provider, or even GitHub Releases" or something like that.

File: src/Wizard.jl [hunk 5]
```diff
@@ -1145,25 +1139,353 @@ function step7(state::WizardState)
...
+
+function authenticate_travis(github_token)
+    resp = HTTP.post("https://api.travis-ci.org/auth/github",
+        body=JSON.json(Dict(
+            "github_token" => github_token
+        )), headers = travis_headers, statusraise = false)
+    JSON.parse(resp.body)["access_token"]
+end
+
+function sync_and_wait_travis(state, repo_name, travis_token)
```
Comment (staticfloat) [inline src/Wizard.jl]: This always takes me a while, since I have 1.6K+ repos on github

File: src/Wizard.jl [hunk 6]
```diff
@@ -1145,25 +1139,353 @@ function step7(state::WizardState)
...
+    old
+end
+
+const travis_headers = Dict(
+    "Accept" => "application/vnd.travis-ci.2+json",
+    "Content-Type" => "application/json",
+    "User-Agent" => "Travis-WHY-YOU-HAVE-BAD-DOCS (https://github.com/travis-ci/travis-ci/issues/5649)"
+)
+
+function authenticate_travis(github_token)
```
Comment (staticfloat) [inline src/Wizard.jl]: Do we know if 2FA screws this up at all?

Comment (Keno) [reply src/Wizard.jl]: No, 2FA works fine. I have it enabled.

File: src/Wizard.jl (hunk 5 above)
Comment (Keno) [reply src/Wizard.jl]: Yeah, it takes forever.

File: src/Wizard.jl [hunk 7]
```diff
@@ -1145,25 +1139,353 @@ function step7(state::WizardState)
...
     """)
 end
 
+mutable struct termios
+	c_iflag::Cint
+	c_oflag::Cint
+	c_cflag::Cint
+	c_lflag::Cint
+	c_line::UInt8
+	c_cc::NTuple{19, UInt8}
```
Comment (ararslan) [inline src/Wizard.jl]: Weird indentation here. (Tabs on these lines and 4 space on the next and everywhere else)

**Pull request merged**